src/java.base/share/classes/java/util/ReverseOrderSortedMapView.java

```diff
@@ -0,0 +1,464 @@
+        return new AbstractCollection<>() {
+            // inherit add(), which throws UOE
+            public Iterator<V> iterator() { return descendingValueIterator(base); }
+            public int size() { return base.size(); }
+        };
+    }
+
+    public Set<Entry<K, V>> entrySet() {
+        return new AbstractSet<>() {
+            // inherit add(), which throws UOE
+            public Iterator<Entry<K, V>> iterator() { return descendingEntryIterator(base); }
+            public int size() { return base.size(); }
+        };
+    }
+
+    // ========== SequencedMap ==========
+
+    public SortedMap<K, V> reversed() {
+        return base;
+    }
+
+    public K firstKey() {
+        return base.lastKey();
+    }
+
+    public K lastKey() {
+        return base.firstKey();
+    }
+
+    public Map.Entry<K, V> firstEntry() {
+        return base.lastEntry();
+    }
+
+    public Map.Entry<K, V> lastEntry() {
+        return base.firstEntry();
+    }
+
+    public Map.Entry<K,V> pollFirstEntry() {
+        return base.pollLastEntry();
+    }
+
+    public Map.Entry<K,V> pollLastEntry() {
+        return base.pollFirstEntry();
+    }
+
+    public V putFirst(K k, V v) {
+        return base.putLast(k, v);
+    }
+
+    public V putLast(K k, V v) {
+        return base.putFirst(k, v);
+    }
+
+    // ========== SortedMap ==========
+
+    public Comparator<? super K> comparator() {
+        return cmp;
+    }
+
+    public SortedMap<K, V> subMap(K fromKey, K toKey) {
+        if (cmp.compare(fromKey, toKey) <= 0) {
+            return new Submap(fromKey, toKey);
+        } else {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    public SortedMap<K, V> headMap(K toKey) {
+        return new Submap(null, toKey);
+    }
+
+    public SortedMap<K, V> tailMap(K fromKey) {
+        return new Submap(fromKey, null);
+    }
+
+    // ========== Infrastructure ==========
+
+    static <K, V> Iterator<K> descendingKeyIterator(SortedMap<K, V> map) {
+        return new Iterator<>() {
+            SortedMap<K, V> root = map;
+            SortedMap<K, V> view = map;
+            K prev = null;
+
+            public boolean hasNext() {
+                return ! view.isEmpty();
+            }
+
+            public K next() {
+                if (view.isEmpty())
+                    throw new NoSuchElementException();
+                K k = prev = view.lastKey();
+                view = root.headMap(k);
+                return k;
+            }
+
+            public void remove() {
+                if (prev == null) {
+                    throw new IllegalStateException();
+                } else {
+                    root.remove(prev);
+                    prev = null;
+                }
+            }
+        };
+    }
+
+    static <K, V> Iterator<V> descendingValueIterator(SortedMap<K, V> map) {
+        return new Iterator<>() {
+            Iterator<K> keyIterator = descendingKeyIterator(map);
```

Comment:
`private final` or better declare it above as a local variable and capture it inside the anonymous class